src/main/java/cms/gov/madie/measure/services/HumanReadableService.java

```diff
@@ -0,0 +1,115 @@
+
+import org.springframework.stereotype.Service;
+import org.apache.commons.lang3.StringUtils;
+
+import cms.gov.madie.measure.exceptions.BundleOperationException;
+import cms.gov.madie.measure.exceptions.ResourceNotFoundException;
+import cms.gov.madie.measure.exceptions.UnsupportedTypeException;
+import cms.gov.madie.measure.factories.ModelValidatorFactory;
+import cms.gov.madie.measure.repositories.ExportRepository;
+import cms.gov.madie.measure.utils.MeasureUtil;
+import gov.cms.madie.models.common.ModelType;
+import gov.cms.madie.models.measure.Export;
+import gov.cms.madie.models.measure.Measure;
+import gov.cms.madie.packaging.utils.PackagingUtility;
+import gov.cms.madie.packaging.utils.PackagingUtilityFactory;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@AllArgsConstructor
+@Service
+public class HumanReadableService {
+
+  private FhirServicesClient fhirServicesClient;
+  private MeasureService measureService;
+  private final ModelValidatorFactory modelValidatorFactory;
+  private final MeasureUtil measureUtil;
+  private final ExportRepository exportRepository;
+
+  public String getHumanReadableWithCSS(String measureId, String username, String accessToken) {
+    Measure measure = measureService.findMeasureById(measureId);
+    if (measure == null) {
+      throw new ResourceNotFoundException("Measure", measureId);
+    }
+
+    // this is temporary until we allow  to get QDM measures HR
+    if (!measure.getModel().contains("QI-Core")) {
+      throw new UnsupportedTypeException(this.getClass().getName(), measure.getModel().toString());
+    }
+
+    return getFhirHR(measure, username, accessToken);
+  }
+
+  protected String getFhirHR(Measure measure, String username, String accessToken) {
+    String humanReadableWithCss = null;
+    boolean saveHR = false;
+    // versioned QI-Core measures will have human readable saved in Export
+    if (measure.getMeasureMetaData() != null && !measure.getMeasureMetaData().isDraft()) {
+      Optional<Export> savedExport = exportRepository.findByMeasureId(measure.getId());
+      if (savedExport.isPresent()) {
+        humanReadableWithCss = savedExport.get().getHumanReadable();
+        if (!StringUtils.isBlank(humanReadableWithCss)) {
+          return humanReadableWithCss;
+        } else {
+          saveHR = true;
+        }
+      } else {
+        saveHR = true;
+      }
+    }
+
+    Measure existingMeasure = validateMeasure(measure);
+
+    String measureBundle =
+        fhirServicesClient.getMeasureBundle(existingMeasure, accessToken, "export");
+
+    humanReadableWithCss = getHRWithCSS(existingMeasure, measureBundle);
+    if (saveHR) {
+      Export export =
+          Export.builder()
+              .measureId(existingMeasure.getId())
+              .measureBundleJson(measureBundle)
+              .humanReadable(humanReadableWithCss)
+              .build();
+      Export savedExport = exportRepository.save(export);
```

Comment:
why are we saving HR for draft measure?

Reply:
@adongare Because in MAT-7243 Joe just implemented saving hr after versioning a qicore measure, but since this is a new feature, and there will be a lot of versioned measures which don't have the hr in db, Joe said we should save it. 
![image](https://github.com/user-attachments/assets/8163c096-d9d4-4cbe-b8e8-ff82b4115e82)

Reply:
got it. thank you for clarification!